### ISSUE TITLE
Support for a global Context

### ISSUE THREAD

**Issue by TheoVC** [codingseb/ExpressionEvaluator#45]
Receiving a list of variables is helpfull, but it would be great if instead of this

`var eval = new ExpressionEvaluator(variables);`

the library could receive an object which would be the context. (similar to what roslyn does)

`var eval = new ExpressionEvaluator(Person1);`

then all the properties, variables and methods would be directly accessible

```
eval.Evaluate(" Name + \" \" + LastName ");
eval.Evaluate(" aPersonMethod() + 10");
```

where Person1 is an instance of the class

```
public class Person{
   public string Name {get;set;}
   public string LastName {get;set;}
   public int aPersonMethod(){
      return 10;
   }
}
```

**Comment by codingseb** [comment]
I will see what it take to do it.
For now I do not have a lot of time for ExpressionEvaluator.
but the idea is good.

**Comment by codingseb** [comment]
I just published version 1.4.6.0 that implement a Context object in addition to the variables dictionary. 
(See  [Context object](https://github.com/codingseb/ExpressionEvaluator/wiki/Variables-and-Functions#context-object) on the wiki)
2 new constructors are available and a property `Context`.

It's an easy modification that allow this and I only made a few tests.
So don't hesitate to open new issues if you find some bugs with this.

For now I close this issue.
Thanks for this idea. I think it's a nice enhancement for ExpressionEvaluator.
